Markdown/Default.sublime-keymap

```diff
@@ -61,33 +48,62 @@
     },
 
     // Auto-pair backticks
-    { "keys": ["`"], "command": "insert_snippet", "args": {"contents": "`$0`"}, "context":
-        [
+    {
+        "keys": ["`"],
+        "command": "insert_snippet",
+        "args": {"contents": "`$0`"},
+        "context": [
             { "key": "setting.auto_match_enabled"},
-            { "key": "selector", "operand": "text.html.markdown - markup.raw" },
             { "key": "selection_empty", "match_all": true },
-            { "key": "preceding_text", "operator": "regex_contains", "operand": "\\S", "match_all": true },
-            { "key": "preceding_text", "operator": "not_regex_contains", "operand": "\\\\$", "match_all": true },
+            { "key": "selector", "operand": "text.html.markdown - markup.raw - meta.code-fence" },
+            { "key": "preceding_text", "operator": "regex_contains", "operand": "(?<![\\\\])([\\\\]{2})*$", "match_all": true }
         ]
     },
     {
         "keys": ["`"],
         "command": "insert_snippet",
         "args": {"contents": "`${0:$SELECTION}`"},
         "context": [
-            { "key": "selector", "operand": "text.html.markdown - markup.raw" },
             { "key": "setting.auto_match_enabled"},
             { "key": "selection_empty", "operand": false, "match_all": true },
+            { "key": "selector", "operand": "text.html.markdown - markup.raw - meta.code-fence" }
         ]
     },
-    // If we have two backticks with the cursor between them, then a backspace removes both
-    { "keys": ["backspace"], "command": "run_macro_file", "args": {"file": "res://Packages/Default/Delete Left Right.sublime-macro"}, "context":
-        [
+    {
+        "keys": ["`"],
+        "command": "move",
+        "args": {"by": "characters", "forward": true},
+        "context": [
+            { "key": "setting.auto_match_enabled" },
+            { "key": "selection_empty", "operand": true, "match_all": true },
+            { "key": "selector", "operand": "text.html.markdown markup.raw - markup.raw.code-fence - meta.code-fence" },
+            { "key": "following_text", "operator": "regex_contains", "operand": "^`", "match_all": true }
+        ]
+    },
+    {
+        "keys": ["backspace"],
+        "command": "run_macro_file",
+        "args": {"file": "res://Packages/Default/Delete Left Right.sublime-macro"},
+        "context": [
             { "key": "setting.auto_match_enabled" },
             { "key": "selection_empty", "match_all": true },
             { "key": "selector", "operator": "equal", "operand": "punctuation.definition.raw.begin.markdown" },
             { "key": "preceding_text", "operator": "regex_contains", "operand": "`$", "match_all": true },
             { "key": "following_text", "operator": "regex_contains", "operand": "^`", "match_all": true }
         ]
     },
+
+    // Add fenced code block
+    {
+        "keys": ["`", "`"],
```

Comment:
I believe bindings with two char keys to be bad UX for the most part, since they consume the first key  until another key is pressed that either triggers a multi-chord binding or matches no bindings and only then are the previously hold-back key presses handled (or a timeout happens). That means after typing the first `` ` `` ST would seemingly do nothing.


It's okay for custom bindings imo or those that don't produce a character, but for character-based bindings I'd avoid them with a `preceding_text` context. It also means they are more flexible since they key presses don't have to be input in consecutively.

Reply:
The reason why nothing is printed when typing the first backtick is not the snippet, but the nature of the caracter itself.

You can also type a plain backtick with all bindings disabled and it will not be added until you hit space or another character it will then be placed on top of.

Reply:
> The reason why nothing is printed when typing the first backtick is not the snippet, but the nature of the caracter itself.

This is incorrect for the non-dead backtick key, as I have it on my keyboard layout (and so does the US layout iirc).

In fact, when I try this locally with ``{ "keys": ["`", "`"], "command": "new_file", },`` it doesn't even work?

Reply:
To clarify my statement, here is a screencast taken with Windows Notepad. Please see the printed key strokes to see how typing a backtick never prints the character without another key stroke following.

![Animation](https://user-images.githubusercontent.com/16542113/106391060-bb8d5c00-63eb-11eb-8fe7-8b0690eded57.png)

In fact `{ "keys": ["`", "`"], "command": "new_file", },` works very well for me. Here the proof for those who don't believe.

![Animation](https://user-images.githubusercontent.com/16542113/106391137-263e9780-63ec-11eb-85b5-bcb83e29f16e.png)

Reply:
This behavior happens because <kbd>\`</kbd> is a dead key in the german keyboard layout.
However, I have also configured my own keyboard layout and removed the dead key for <kbd>\`</kbd>, but I still have no problems when I try the auto-pair keybinding from this PR, i.e. it works fine and the first \` is still inserted as expected.

Reply:
I think the binding not working is just an ST bug with my configuration. If it works for you, then it's all good. The binding *is* triggered for ``["`"]``, fwiw.